packages/components/scripts/wrapper-generator/VueWrapperGenerator.ts

```diff
@@ -0,0 +1,144 @@
+import { AbstractWrapperGenerator } from './AbstractWrapperGenerator';
+import { getComponentMeta } from '@porsche-design-system/shared';
+import type { TagName } from '@porsche-design-system/shared';
+import type { ExtendedProp } from './DataStructureBuilder';
+import { camelCase, pascalCase } from 'change-case';
+
+export class VueWrapperGenerator extends AbstractWrapperGenerator {
+  protected packageDir = 'components-vue';
+  protected projectDir = 'vue-wrapper';
+
+  public getComponentFileName(component: TagName): string {
+    return `${pascalCase(component.replace('p-', ''))}Wrapper.vue`;
+  }
+
+  public generateImports(component: TagName, extendedProps: ExtendedProp[], nonPrimitiveTypes: string[]): string {
+    const hasEventProps = extendedProps.some(({ isEvent }) => isEvent);
+    const hasProps = getComponentMeta(component).props;
+
+    const vueImports = ['onMounted', 'onUpdated', 'ref'];
+    const importsFromVue = hasProps ? `import { ${vueImports.join(', ')} } from 'vue';` : '';
+
+    const utilsImports = [
+      'getPrefixedTagName',
+      ...(hasProps ? ['syncProperties'] : []),
+      ...(hasEventProps ? ['addEventListenerToElementRef'] : []),
+    ];
+    const importsFromUtils = `import { ${utilsImports.join(', ')} } from '../../utils';`;
+
+    const importsFromTypes = nonPrimitiveTypes.length
+      ? `import type { ${nonPrimitiveTypes.join(', ')} } from '../types';`
+      : '';
+
+    return `<script setup lang="ts">
+  ${[importsFromVue, importsFromUtils, importsFromTypes].filter((x) => x).join('\n  ')}`;
```

Comment:
Do indentation in a single spot or if not possible, at least reuse a variable for the spaces.